// blog/blog.js
"use strict";

var User = require('./lib/user');
var Article = require('./lib/article');

var Blog = module.exports = function() {
	this.config = {}; 
};

Blog.prototype.init = function(callback) {
	if (callback)
		callback(null, 'init_function().');
};

Blog.prototype.login = function(callback) {
	var user = new User();
	user.login(function(err, echo){
		if (err)
			if (callback)
				callback(err);	
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.articles = function(callback) {
	var article = new Article();
	article.articles(function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.GetArticles = function(callback) {
	var article = new Article();
	article.GetArticles(function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.GetArticle = function(ArticleID, callback) {
	var article = new Article();
	article.GetArticle(ArticleID, function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.AddArticle = function(callback) {
	var article = new Article();
	article.AddArticle(function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.UpdateArticle = function(ArticleID, callback) {
	var article = new Article();
	article.UpdateArticle(ArticleID, function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.DeleteArticle = function(ArticleID, callback) {
	var article = new Article();
	article.DeleteArticle(ArticleID, function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.Comments = function(ArticleID, callback) {
	var article = new Article();
	article.Comments(ArticleID, function(err, echo){
		if (callback)
			callback(null, echo);
	});
};

Blog.prototype.AddComment = function(ArticleID, callback) {
	var article = new Article();
	article.AddComment(ArticleID, function(err, echo){
		if (callback)
			callback(null, echo);
	});
};
