// blog/lib/article.js
"use strict";

var Article = module.exports = function() {
	this.config = {}; 
};

Article.prototype.articles = function(callback) {
	if (callback)
		callback(null, 'Get all articles.');
};

Article.prototype.GetArticles = function(callback) {
	if (callback)
		callback(null, 'Get all articles by user.');
};

Article.prototype.GetArticle = function(ArticleID, callback) {
	if (callback)
		callback(null, 'Get article by Article ID');
};

Article.prototype.AddArticle = function(callback) {
	if (callback)
		callback(null, 'Add a Article by user.');
};

Article.prototype.UpdateArticle = function(ArticleID, callback) {
	if (callback)
		callback(null, 'Update a Article');
};

Article.prototype.DeleteArticle = function(ArticleID, callback) {
	if (callback)
		callback(null, 'Delete a Article');
};

Article.prototype.Comments = function(ArticleID, callback) {
	if (callback)
		callback(null, 'Get all comments by Article ID.');
};

Article.prototype.AddComment = function(ArticleID, callback) {
	if (callback)
		callback(null, 'Add a new comment by Article ID');
};
